lib/phoenix_live_view/tag_engine.ex

```diff
@@ -443,6 +444,14 @@ defmodule Phoenix.LiveView.TagEngine do
     |> update_subengine(:handle_expr, [marker, expr])
   end
 
+  defp handle_token({:body_expr, value, %{line: line, column: column}}, state) do
+    quoted = Code.string_to_quoted!(value, line: line, column: column, file: state.file)
```

Comment:
For this template:

```heex
{if true do}
```

we currently receive this error:

```elixir
** (TokenMissingError) token missing on iex:5:11:
    error: missing terminator: end
    │
  5 │ if true do
    │         │ └ missing closing delimiter (expected "end")
    │         └ unclosed delimiter
    │
    └─ iex:5:11
```

which is great but maybe there's still room to improve?

If we changed `Code.string_to_quoted!` to `Code.string_to_quoted` we could maybe improve the error message, telling people to use `:if` or  `<%= ... %>`. The error tuple is quite expressive:

```elixir
iex> Code.string_to_quoted("if true do")
{:error,
 {[
    opening_delimiter: :do,
    expected_delimiter: :end,
    line: 1,
    column: 9,
    end_line: 1,
    end_column: 11
  ], "missing terminator: end", ""}}
```

and I see it's like this since Elixir 1.16. Let me know if you'd like to pursue this and I'm happy to send a patch.